### ISSUE TITLE
Regression after #802 fix: munin-cgi-graph log spamming: [PERL WARNING] Use of uninitialized value $size_x in string eq at /usr/lib/munin/cgi/munin-cgi-graph line 453

### ISSUE THREAD

**Issue by carnil** [munin-monitoring/munin#804]
After the #802 fix, logs for munin-cgi-graph are spammed with uniitialized value warnings for the various parameters, depending which sites are visited.

```
[...]
2017/03/02 06:53:56 [PERL WARNING] Use of uninitialized value $size_x in string eq at /usr/lib/munin/cgi/munin-cgi-graph line 453.
[...]
```

Reported as well in Debian as https://bugs.debian.org/856536

**Comment by mejo-** [comment]
To my understanding, only *numbers* are valid values for `size_x`, `size_y`, `upper_limit` and `lower_limit`. Why not use [`Scalar::Util::looks_like_number()`](http://perldoc.perl.org/Scalar/Util.html#looks_like_number) and do something like:

```
use Scalar::Util qw(looks_like_number);
[...]
    push @params, "--lower_limit", $lower_limit if looks_like_number($lower_limit) > 0;
```

Or even better, add some similar code to check for valid values (i.e. *numbers*) to function [`verify_parameters()`](https://github.com/munin-monitoring/munin/blob/stable-2.0/master/_bin/munin-cgi-graph.in#L356_L402), which would be the correct place to sanitize the parameters, no?

**Comment by steveschnepp** [comment]
For `looks_like_number()` I didn't want to depend on another function in here. Well, NIH struck again :(

For the `verify_parameters()` it doesn't fill the bill, as it only validates those. It doesn't set them.

**Comment by mejo-** [comment]
Looks good to me now. Do you intend to release another 2.0 release to fix the regression?

Regarding `verify_parameters()`: why not error out in case of invalid value instead of ignoring it? But that's out of scope here.

**Comment by sumpfralle** [comment]
The fix is merged in 2.x and assumably not relevant for master. Closing.
